### PR TITLE
fix erlang build version, dialyzer warnings, compile warnings

### PR DESCRIPTION
**Pull request by lbakerchef**
### Description

Use the correct version of Erlang for the unit tests in the verify pipeline.
Fix dialyzer warnings.
Fix compiler warnings.

https://buildkite.com/chef/chef-chef-server-master-verify/builds/820#c441a902-a648-439f-b532-f1b053ba4635
https://buildkite.com/chef/chef-chef-server-master-omnibus-adhoc/builds/693#494e0bd8-b1b2-4bf0-a6ce-00bf871c443d

### Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussions that are relevant]

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

File: src/chef-mover/_checkouts/oc_erchef/rebar.config [hunk 1]
```diff
@@ -15,8 +15,8 @@
   {git,"git://github.com/opscode/chef_authn.git",
           "5d0577f693404898d6f05dbcbb38ffe2cc36c15d"}},
  {couchbeam,".*",
-  {git,"git://github.com/opscode/couchbeam.git",
-         {branch, "lbaker/fixes_for_erlang19"}}},
+  {git,"git://github.com/chef/couchbeam.git",
```
Comment (PrajaktaPurohit) [inline src/chef-mover/_checkouts/oc_erchef/rebar.config]: opscode and chef both redirect to the same repo in chef-boneyard (https://github.com/chef-boneyard/couchbeam)

File: src/oc_erchef/rebar.lock [hunk 2]
```diff
@@ -36,8 +36,8 @@
        {ref,"c2a6b925ed94ab8a28ff49375782834033919190"}},
   0},
  {<<"ej">>,
-  {git,"git://github.com/seth/ej.git",
-       {ref,"40821f7ab3d8746e8573e8542ad452d2b346ed7f"}},
+  {git,"git://github.com/lbakerchef/ej.git",
```
Comment (PrajaktaPurohit) [inline src/oc_erchef/rebar.lock]: lets point this to the repo in the chef org.

File: src/chef-mover/_checkouts/oc_erchef/rebar.config [hunk 3]
```diff
@@ -24,8 +24,8 @@
   {git,"git://github.com/seth/edown.git",
          "754be25f71a04099c83f3ffdff268e70beeb0021"}},
  {ej,".*",
-  {git,"git://github.com/lbakerchef/ej.git",
-          "40821f7ab3d8746e8573e8542ad452d2b346ed7f"}},
+  {git,"git://github.com/chef/ej.git",
```
Comment (PrajaktaPurohit) [inline src/chef-mover/_checkouts/oc_erchef/rebar.config]: you are already doing that!

File: src/chef-mover/rebar.config [hunk 4]
```diff
@@ -62,8 +62,8 @@
 
     %%% Oldies
     % {ej, ".*",
-    %  {git,"git://github.com/seth/ej.git",
-    %    "132a9a3c0662a2377eaf7ebee694a496a0957160"}},
+    %  {git,"git://github.com/chef/ej.git",
```
Comment (PrajaktaPurohit) [inline src/chef-mover/rebar.config]: We can probably just get rid of this section.

File: src/chef-mover/src/mv_oc_chef_authz.hrl [hunk 5]
```diff
@@ -16,10 +16,10 @@
 -type oc_authz_id() :: <<_:256>>.
 
 -record(authz_group, {actors = [] :: actor_list(),
-		      groups = [] :: group_list()}).
+		              groups = [] :: group_list()}).
```
Comment (PrajaktaPurohit) [inline src/chef-mover/src/mv_oc_chef_authz.hrl]: should be alined under the `actors`

File: src/chef-mover/_checkouts/oc_erchef/rebar.config (hunk 1 above)
Comment (lbakerchef) [reply src/chef-mover/_checkouts/oc_erchef/rebar.config]: Yes.  I believe Mark said I should use the 'chef' name for all of those just to be consistent, so I changed them.

File: src/oc_erchef/rebar.lock (hunk 2 above)
Comment (lbakerchef) [reply src/oc_erchef/rebar.lock]: This was already changed previously.  Not sure why it's showing incorrectly here.

File: src/chef-mover/src/mv_oc_chef_authz.hrl (hunk 5 above)
Comment (lbakerchef) [reply src/chef-mover/src/mv_oc_chef_authz.hrl]: Weird.  It was aligned perfectly when I brought it up in vim.  Then I discovered some weird invisible characters (not spaces) which were apparently throwing off the alignment here.

File: src/chef-mover/rebar.config (hunk 4 above)
Comment (lbakerchef) [reply src/chef-mover/rebar.config]: Done.

File: src/chef-mover/src/mv_oc_chef_authz.hrl (hunk 5 above)
Comment (lbakerchef) [reply src/chef-mover/src/mv_oc_chef_authz.hrl]: Fixed.

File: src/oc_bifrost/rebar.config [hunk 6]
```diff
@@ -20,7 +20,7 @@
     {edown, ".*",
         {git, "git://github.com/uwiger/edown.git",                    {branch, "master"}}},
     {ej, ".*",
-        {git, "git://github.com/seth/ej.git",                         {branch, "master"}}},
+        {git, "git://github.com/chef/ej.git",                         {branch, "lbaker/fixes-for-erlang19"}}},
```
Comment (markan) [inline src/oc_bifrost/rebar.config]: Should this be merged? If not, is there an issue to get this merged and update deps?

File: src/bookshelf/include/bksw_obj.hrl [hunk 7]
```diff
@@ -1,8 +1,8 @@
--record(object, {path :: binary(),
```
[the remark this answers is not in the document]
Comment (lbakerchef) [reply src/bookshelf/include/bksw_obj.hrl]: I'll definitely get this fixed-up to your liking.  We'll talk more after the three-day holiday.

File: src/oc_bifrost/rebar.config (hunk 6 above)
Comment (PrajaktaPurohit) [reply src/oc_bifrost/rebar.config]: https://github.com/chef/chef-server/issues/1822

File: src/chef-mover/src/mv_oc_chef_authz.hrl (hunk 5 above)
Comment (PrajaktaPurohit) [reply src/chef-mover/src/mv_oc_chef_authz.hrl]: it still seems to be misaligned.

**Pull request merged**